### ISSUE TITLE
add `--schemaURL` option to `wgc subgraph publish`

### ISSUE THREAD

**Issue by kappasims** [wundergraph/cosmo#662]
### Is your feature request related to a problem? Please describe.

For code-first GraphQL implementations, a physical schema file doesn't necessarily exist in the project. Rather, it is more common for a server-generated schema to exist at a specific endpoint. When my schema changes, I currently must download this file manually and copy it to my project, then run the publish command, or script this process.

### Describe the solution you'd like

It seems like it would be fairly trivial to provide a command line option for the URL at which a schema resides, make a curl or similar request, and then pipe the output to a temporary file, at which point you could hook into to the existing publish flow.

### Describe alternatives you've considered

Scripting this manual process

**Comment by jensneuse** [comment]
How would you handle authentication? What if the URL is not reachable from where the wgc command runs? What if introspection is disabled? Would these be cases where this approach simply won't work?

Another thought. When you publish a schema file, it's usually part of a repo and checked into git. With introspection, it's possible that errors occur.

**Comment by kappasims** [comment]
First, I think it's important that I clarify that I'm not proposing creating some link between a schema's URL and Cosmo, I'm simply asking for the ability to download the schema from a URL, store it to a temporary location, and then publish to Cosmo, without having to have a physical file in my project. This is easier for both local development and GitHub Actions, which I'm currently having to write scripts manually to do.

Authentication: It would be okay to assume the URL is unauthenticated.
Failure scenario: Since this is a one-time DL and publish, if the URL is not returning a 200 or expected status code then it is okay to abort.

Since the schema is defined in the code, the code itself can be rolled back to a specific point in time and deployed or spun up locally, and the schema can be downloaded and published at that time. I am actually specifically trying to avoid a case where a checked-in schema differs from the schema generated by the server. That is the worst kind of error.

**Comment by jensneuse** [comment]
Thanks for your input. Do you think it would make sense to also add this command for "wgc check", or only publish?

**Comment by kappasims** [comment]
@jensneuse Yes it does absolutely! Sorry if I overlooked that (and am overlooking any other applicable commands with the --schema option)

**Comment by kappasims** [comment]
One other item I wanted to clarify because it may come up in discussion: If introspection is disallowed in the production environment, is this feature even useful? The answer is yes, because without this feature the alternative is the same: spin up a docker container in dev mode with introspection on, grab the schema, and publish it.

**Comment by endigma** [comment]
Hi @kappasims , closing this as stale, please feel free to reopen if it's still relevant to you!
